File: Backend/src/main/java/app/carro/entity/Carro.java
package app.carro.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@NoArgsConstructor
@Entity
public class Carro {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
    private String marca;
    private String modelo;
    private Integer ano;
    private String cor;
    private Double preco;

}

File: carros_frontend/src/app/models/carro.ts
import { Acessorio } from "./acessorio";
import { Marca } from "./marca";

export class Carro {
    id!: number;
    marca!: Marca;
    modelo!: string;
    cor!: string;
    ano!: number;
    preco!: number;
    acessorio: Acessorio[] = [];

    constructor(id: number, marca: Marca, modelo: string, cor: string, ano: number, preco: number){
        this.id = id;
        this.marca = marca;
        this.modelo = modelo;
        this.cor = cor;
        this.ano = ano;
        this.preco = preco;
    }
}
